nilearn/surface/tests/test_surface.py

```diff
@@ -39,42 +50,43 @@
 datadir = Path(__file__).resolve().parent / "data"
 
 
-class MeshLikeObject:
-    """Class with attributes coordinates \
-       and faces to be used for testing purposes.
-    """
-
-    def __init__(self, coordinates, faces):
-        self._coordinates = coordinates
-        self._faces = faces
-
-    @property
-    def coordinates(self):
-        return self._coordinates
-
-    @property
-    def faces(self):
-        return self._faces
-
-
-class SurfaceLikeObject:
-    """Class with attributes mesh and data to be used for testing purposes."""
-
-    def __init__(self, mesh, data):
-        self._mesh = mesh
-        self._data = data
-
-    @classmethod
-    def fromarrays(cls, coordinates, faces, data):
-        return cls(MeshLikeObject(coordinates, faces), data)
+def test_check_mesh():
+    mesh = check_mesh("fsaverage5")
+    assert mesh is check_mesh(mesh)
+    with pytest.raises(ValueError):
+        check_mesh("fsaverage2")
+    mesh.pop("pial_left")
+    with pytest.raises(ValueError):
+        check_mesh(mesh)
+    with pytest.raises(TypeError):
+        check_mesh(load_surf_mesh(mesh["pial_right"]))
 
-    @property
-    def mesh(self):
-        return self._mesh
 
-    @property
-    def data(self):
-        return self._data
+def test_check_mesh_and_data(rng):
+    coords, faces = generate_surf()
+    mesh = InMemoryMesh(coords, faces)
```

Comment:
```suggestion
    mesh = generate_surf()
```

I think you already modified generate_surf to return a InMemoryMesh